controllers/mongodbcluster_controller.go

```diff
@@ -472,3 +528,159 @@ func (r *MongoDBClusterReconciler) reconcileAtlasFirewall(ctx context.Context, m
+	if scheduledAutoscaling != nil && scheduledAutoscaling.Enabled {
+		var foundUp gocron.Job
+
+		var foundDown gocron.Job
+
+		jobs := r.Scheduler.Jobs()
+		if jobs != nil {
+			for _, job := range jobs {
+				if job.Tags()[0] == mongodbClusterCR.Name && job.Tags()[2] == "up" {
+					foundUp = job
+				} else if job.Tags()[0] == mongodbClusterCR.Name && job.Tags()[2] == "down" {
+					foundDown = job
+				}
+			}
+		} else {
+			return fmt.Errorf("list of jobs is nil, wtf? Did the scheduler not initialize?")
+		}
+
+		// Is this client gonna expire on me? Or is it eternal? ChatGPT says it wont expire, but I don't trust it.
+		client, atlasGroupID, err := getAtlasClientFromSecret(secret)
+		if err != nil {
+			logger.Error(err, "Couldn't get a client for Atlas")
+			return err
+		}
+
+		clusterName, err := getClusterNameFromHostTemplate(ctx, client, atlasGroupID, mongodbClusterCR.Spec.HostTemplate)
+		if err != nil {
+			logger.Error(err, "Couldn't find cluster in Atlas")
+			return err
+		}
+
+		clusterDetails, response, err := client.Clusters.Get(ctx, atlasGroupID, clusterName)
+		if err != nil || response.StatusCode != http.StatusOK {
+			logger.Error(err, "Couldn't get cluster details from Atlas")
+			return err
+		}
+
+		if foundDown == nil || foundDown.Tags()[1] != scheduledAutoscaling.ScaleDownCronExpression+scheduledAutoscaling.LowTier {
+			if foundDown != nil {
+				logger.Info("Removing outdated downscaling job for " + mongodbClusterCR.Name)
+
+				err = r.Scheduler.RemoveJob(foundDown.ID())
+				if err != nil {
+					logger.Error(err, "Error removing outdated downscaling job")
+					return err
+				}
+			}
+
+			logger.Info("Creating scheduled downscaling job for " + mongodbClusterCR.Name + " with expression " + scheduledAutoscaling.ScaleDownCronExpression + " to " + scheduledAutoscaling.LowTier)
+
+			_, err = r.Scheduler.NewJob(
+				gocron.CronJob(scheduledAutoscaling.ScaleDownCronExpression, false),
+				gocron.NewTask(
+					func() error {
+						logger.Info("Scaling down " + mongodbClusterCR.Name + " to " + scheduledAutoscaling.LowTier)
+
+						r.EventRecorder.Event(mongodbClusterCR, corev1.EventTypeNormal, "Scaling", "Scaling down to "+scheduledAutoscaling.LowTier)
+
+						_, response, err := client.Clusters.Update(ctx, atlasGroupID, clusterName, &mongodbatlas.Cluster{
+							ProviderSettings: &mongodbatlas.ProviderSettings{
+								ProviderName:     "AWS",
+								InstanceSizeName: scheduledAutoscaling.LowTier,
+								RegionName:       clusterDetails.ProviderSettings.RegionName,
+							},
+						})
+
+						if err != nil || response.StatusCode != http.StatusOK {
+							logger.Error(err, "Couldn't scale down "+mongodbClusterCR.Name)
+							r.EventRecorder.Event(mongodbClusterCR, corev1.EventTypeWarning, "Scaling", "Failed to scale down to "+scheduledAutoscaling.LowTier)
+
+							// Flip a metric so we can alert on this. This one is a warning.
+							ScalingDownErrorGauge.WithLabelValues(mongodbClusterCR.Name).Inc()
```

Comment:
same comment as above.

also, should the message here include err.Error() if it's non-nil? that way we know from describe what went wrong instead of jumping back to logs?